File: 0049 group-anagrams/solution.ts
function groupAnagrams(strs: string[]): string[][] {
    let map = new Map();
    for(const str of strs) {
        let key = str.split('').sort().join('');
        if(!map.has(key)) {
            map.set(key,[]);
        }
        map.get(key).push(str);
    }
    let ans = [];
    for(let [_,val] of map) {
        ans.push(val);
    }
    return ans;
};

File: 0049 group-anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<string>> map;
        for(string s:strs) {
            string key = s;
            sort(key.begin(), key.end());
            map[key].push_back(s);
        }
        vector<vector<string>> ans;
        for(auto& pair : map) {
            ans.push_back(pair.second);
        }
        return ans;
    }
};
